### ISSUE TITLE
Event lag side effects

### ISSUE THREAD

**Issue by sdizazzo** [sdizazzo/pyaltitude#4]
There is lag between when the event happens in the game for the player (powerupPickup) as it is output in the `log.txt`, read, parsed and handled by pyaltitude, and then corresponding events written back out to `command.txt` and processed by the Altitude server.  Even when only parts of this process are followed, side effects can develop.  Depending on how many events you throw at the `command.txt` how quickly, it can get behind by a littlle (half second) to ALOT (30 seconds or more). During that time any number of things can become out of sync or messages and events will come late.  One example that shows up as a bad bug, is when tracking players who are logged into the server and which events correspond to which player numbers.  If a player logs off, but the server is behind in processing events, an event might still come for their player number.  In that case it will ask the Server() object for that player number and raise an exception because it has already been removed.  I could (and have) then, just added checks to see if the player exists, but besides being bad form, I also realized that during that time another player could have also just taken over that player number.  In that case, a different player would be getting the events intended for the player that already logged off!  I guess its kind of a very slow race condition.

**Comment by sdizazzo** [comment]
I believe I finally found the issue with my client tracking issues and I think you are correct, it's not an issue of lag.  It's just the `playerInfoEv` event that comes right after the `clientRemove` event that is the issue.  Now that I have logging its easy to see.  It has the `leaving` attr set to `True`, so I just ignore it and everything seems to work fine so far.  DOH!

**Comment by sdizazzo** [comment]
I had other issues because of thread pools on each Server() object as well.  When many events would come in at a time, there's no guarantee what order they would be executed in,so when a player entered or exited, sometimes the `playerInfoEv` would be executed in the pool before the `clientAdd`.  Other funky business was happening as well.  I finally removed the the thread pools and it seems to be much more stable. Everything is guaranteed to execute in order.  Took quite a while to get the whole thing right!

In regards to the lag, its improved since I forked Xal's code, but is a different issue.  I believe you still could use a thread pool on each servers if needed for the events that don't need to execute in order like kills, pushes, etc, and then offload the `clientAdd`,  `clientRemove`, etc to a dedicated queue.
